src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -590,13 +572,138 @@ BoundExpression convertArrayElement(BoundExpression element, TypeSymbol elementT
+            {
+                var analyzedArguments = AnalyzedArguments.GetInstance();
+                collectionCreation = BindClassCreationExpression(syntax, namedType.Name, syntax, namedType, analyzedArguments, diagnostics);
+                collectionCreation.WasCompilerGenerated = true;
+                analyzedArguments.Free();
+            }
+            else if (targetType is TypeParameterSymbol typeParameter)
+            {
+                var arguments = AnalyzedArguments.GetInstance();
+                collectionCreation = BindTypeParameterCreationExpression(syntax, typeParameter, arguments, initializerOpt: null, typeSyntax: syntax, wasTargetTyped: true, diagnostics);
+                arguments.Free();
+            }
+            else
+            {
+                collectionCreation = new BoundBadExpression(syntax, LookupResultKind.NotCreatable, ImmutableArray<Symbol?>.Empty, ImmutableArray<BoundExpression>.Empty, targetType);
+            }
+
+            var collectionInitializerAddMethodBinder = this.WithAdditionalFlags(BinderFlags.CollectionInitializerAddMethod);
+            var builder = ArrayBuilder<BoundExpression>.GetInstance(node.Initializers.Length);
+            foreach (var element in node.Initializers)
+            {
+                var result = BindCollectionInitializerElementAddMethod(
+                    element.Syntax,
+                    ImmutableArray.Create(element),
+                    hasEnumerableInitializerType,
+                    collectionInitializerAddMethodBinder,
+                    diagnostics,
+                    implicitReceiver);
+                result.WasCompilerGenerated = true;
+                builder.Add(result);
+            }
+            return new BoundCollectionInitializerCollectionLiteralExpression(
+                syntax,
+                collectionCreation,
+                naturalTypeOpt: null, // PROTOTYPE: Support natural type.
+                wasTargetTyped: true, // PROTOTYPE: Support natural type.
+                implicitReceiver,
+                builder.ToImmutableAndFree(),
+                targetType)
+            { WasCompilerGenerated = wasCompilerGenerated };
+        }
+
+        private BoundExpression? ConvertCollectionLiteralExpressionToNaturalType(
+            BoundUnconvertedCollectionLiteralExpression node,
+            BindingDiagnosticBag diagnostics,
+            bool returnNullIfNoCommonType) // PROTOTYPE: Remove.
+        {
+            bool hasErrors = node.HasErrors;
+            var syntax = (CollectionCreationExpressionSyntax)node.Syntax;
+            var location = syntax.OpenBracketToken.GetLocation();
+
+            var useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
+            // PROTOTYPE: Confirm with LDM that we use the element expressions rather than
+            // the element types to infer best common type.
+            var commonType = BestTypeInferrer.InferBestType(node.Initializers, Conversions, ref useSiteInfo, out _);
```

Comment:
yes, using hte expressions is my expectation here.  i would also expect this to match what we do for `new[] { x, y, z }`